### PR TITLE
[FLINK-29149] Migrate E2E tests to catalog-based and enable E2E tests for Flink 1.14

### PR DESCRIPTION
**Pull request by LadyForest**
- Get rid of `ManagedTableFactory`
- Use Kafka connector to replace filesystem connector, since for Flink 1.14, [File system sources for streaming is still under development. In the future, the community will add support for common streaming use cases, i.e., partition and directory monitoring](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/connectors/table/filesystem/)

### CODE REVIEW

File: flink-table-store-format/src/main/java/org/apache/flink/table/store/format/parquet/ParquetInputFormatFactory.java [hunk 1]
```diff
@@ -38,34 +40,35 @@ public static BulkFormat<RowData, FileSourceSplit> create(
             RowType producedRowType,
             TypeInformation<RowData> producedTypeInfo,
             boolean isUtcTimestamp) {
-        Class<?> formatClass = null;
+        Class<?> formatClass;
```
Comment (LadyForest) [inline flink-table-store-format/src/main/java/org/apache/flink/table/store/format/parquet/ParquetInputFormatFactory.java:43]: Need to rebase the master once FLINK-29506 is merged.

File: flink-table-store-e2e-tests/src/test/java/org/apache/flink/table/store/tests/HiveE2eTest.java [hunk 2]
```diff
@@ -33,6 +34,7 @@
  * <p>NOTE: This test runs a complete Hadoop cluster in Docker, which requires a lot of memory. If
  * you're running this test locally, make sure that the memory limit of your Docker is at least 8GB.
  */
+@DisabledIfSystemProperty(named = "flink.version", matches = "1.14.*")
```
Comment (LadyForest) [inline flink-table-store-e2e-tests/src/test/java/org/apache/flink/table/store/tests/HiveE2eTest.java:37]: Note: this class is not disabled if running it single from IDEA. See https://youtrack.jetbrains.com/issue/IDEA-194561

File: flink-table-store-format/src/main/java/org/apache/flink/table/store/format/parquet/ParquetInputFormatFactory.java (hunk 1 above)
Comment (LadyForest) [reply flink-table-store-format/src/main/java/org/apache/flink/table/store/format/parquet/ParquetInputFormatFactory.java]: > Need to rebase the master once [FLINK-29506](https://issues.apache.org/jira/browse/FLINK-29506) is merged.

done

**Review by JingsongLi — APPROVED** [review]
Thanks @LadyForest for the contribution!
Looks good to me!

**Pull request merged**